tests/warp/warp_test.go

```diff
@@ -31,176 +44,160 @@ import (
-	warpTableEntries                      = []ginkgo.TableEntry{
-		ginkgo.Entry("SubnetA -> SubnetB", func() *warpTest {
-			return newWarpTest(context.Background(), subnetA, fundedKey, subnetB, fundedKey)
-		}),
-		ginkgo.Entry("SubnetA -> SubnetA", func() *warpTest {
-			return newWarpTest(context.Background(), subnetA, fundedKey, subnetA, fundedKey)
-		}),
-		ginkgo.Entry("SubnetA -> C-Chain", func() *warpTest {
-			return newWarpTest(context.Background(), subnetA, fundedKey, cChainSubnetDetails, fundedKey)
-		}),
-		ginkgo.Entry("C-Chain -> SubnetA", func() *warpTest {
-			return newWarpTest(context.Background(), cChainSubnetDetails, fundedKey, subnetA, fundedKey)
-		}),
-		ginkgo.Entry("C-Chain -> C-Chain", func() *warpTest {
-			return newWarpTest(context.Background(), cChainSubnetDetails, fundedKey, cChainSubnetDetails, fundedKey)
-		}),
-	}
+const (
+	subnetAName = "warp-subnet-a"
+	subnetBName = "warp-subnet-b"
 )
 
-var _ = ginkgo.DescribeTable("[Warp]", func(gen func() *warpTest) {
-	w := gen()
-
-	log.Info("Sending message from A to B")
-	w.sendMessageFromSendingSubnet()
+var (
+	flagVars *e2e.FlagVars
 
-	log.Info("Aggregating signatures via API")
-	w.aggregateSignaturesViaAPI()
+	repoRootPath = tests.GetRepoRootPath("tests/warp")
 
-	log.Info("Aggregating signatures via p2p aggregator")
-	w.aggregateSignatures()
+	genesisPath = filepath.Join(repoRootPath, "tests/precompile/genesis/warp.json")
 
-	log.Info("Delivering addressed call payload to receiving subnet")
-	w.deliverAddressedCallToReceivingSubnet()
+	subnetA, subnetB, cChainSubnetDetails *Subnet
 
-	log.Info("Delivering block hash payload to receiving subnet")
-	w.deliverBlockHashPayload()
+	testPayload = []byte{1, 2, 3}
+)
 
-	log.Info("Executing HardHat test")
-	w.executeHardHatTest()
+func init() {
+	// Configures flags used to configure tmpnet (via SynchronizedBeforeSuite)
+	flagVars = e2e.RegisterFlags()
+}
 
-	log.Info("Executing warp load test")
-	w.warpLoad()
-}, warpTableEntries)
+// Subnet provides the basic details of a created subnet
+type Subnet struct {
+	// SubnetID is the txID of the transaction that created the subnet
+	SubnetID ids.ID
+	// For simplicity assume a single blockchain per subnet
+	BlockchainID ids.ID
+	// Key funded in the genesis of the blockchain
+	PreFundedKey *ecdsa.PrivateKey
+	// ValidatorURIs are the base URIs for each participant of the Subnet
+	ValidatorURIs []string
+}
 
 func TestE2E(t *testing.T) {
 	gomega.RegisterFailHandler(ginkgo.Fail)
 	ginkgo.RunSpecs(t, "subnet-evm warp e2e test")
 }
 
-func toWebsocketURI(uri string, blockchainID string) string {
-	return fmt.Sprintf("ws://%s/ext/bc/%s/ws", strings.TrimPrefix(uri, "http://"), blockchainID)
-}
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
+	// Run only once in the first ginkgo process
+
+	chainConfig := tmpnet.FlagsMap{
+		"log-level":        "debug",
+		"warp-api-enabled": true,
+	}
+
+	nodes := utils.NewTmpnetNodes(tmpnet.DefaultNodeCount)
```

Comment:
I think this runs the network with 2 nodes, right? It was 10 nodes before (5 for each subnet). Is there anything I miss?

IMHO `DefaultNodeCount` in tmpnet package should be 5 but I guess that's another story.

Reply:
I think for any functional testing we should stick to the minimum number of nodes.